src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -192,6 +203,27 @@ public AnomalyDetector(
         this.categoryFields = categoryFields;
         this.user = user;
         this.detectorType = isMultientityDetector(categoryFields) ? MULTI_ENTITY.name() : SINGLE_ENTITY.name();
+        this.resultIndex = Strings.trimToNull(resultIndex);
+        String errorMessage = validateResultIndex(this.resultIndex);
+        if (errorMessage != null) {
+            throw new IllegalArgumentException(errorMessage);
+        }
+    }
+
+    public static String validateResultIndex(String resultIndex) {
+        if (resultIndex == null) {
+            return null;
+        }
+        if (!resultIndex.startsWith(CUSTOM_RESULT_INDEX_PREFIX)) {
+            return INVALID_RESULT_INDEX_PREFIX;
+        }
+        if (resultIndex.length() > MAX_RESULT_INDEX_NAME_SIZE) {
+            return INVALID_RESULT_INDEX_NAME_SIZE;
+        }
+        if (!resultIndex.matches(RESULT_INDEX_NAME_PATTERN)) {
+            return INVALID_CHAR_IN_RESULT_INDEX_NAME;
```

Comment:
Can this branch ever be true when resultIndex.startsWith(CUSTOM_RESULT_INDEX_PREFIX) is true?

Reply:
I think you mean this test case, check `AnomalyDetectorTests#testValidateResultIndex`
```
    errorMessage = AnomalyDetector.validateResultIndex(CUSTOM_RESULT_INDEX_PREFIX + "abc#");
    assertEquals(INVALID_CHAR_IN_RESULT_INDEX_NAME, errorMessage);
```